File: leetcode/1easy/이경호/3289.cpp
#include <vector>
#include <set>

using namespace std;

class Solution {
public:
  vector<int> getSneakyNumbers(vector<int>& nums) {
    vector<int> answer;
    set<int> exists;
    for (int num : nums) {
      if (exists.find(num) != exists.end()) {
        answer.push_back(num);
      }
      else {
        exists.insert(num);
      }
    }
    return answer;
  }
};

File: leetcode/1easy/위성원/3289.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getSneakyNumbers = function (nums) {
  const counts = new Array(103).fill(0);
  const answer = []
  nums.forEach((num) => {
    const count = counts[num];

    if (count > 0) {
      answer.push(num)
    }

    counts[num] = count + 1;
  })

  return answer
};